src/main.cpp

```diff
@@ -679,9 +679,6 @@ bool IsStandardTx(const CTransaction& tx, string& reason)
         else if ((whichType == TX_MULTISIG) && (!fIsBareMultisigStd)) {
             reason = "bare-multisig";
             return false;
-        } else if (txout.IsDust(::minRelayTxFee)) {
```

Comment:
Yes, that's what I intended. I don't see any problem that this causes; there's no particular reason why you should treat a 0 value txout differently than, say, a 1 zatoshi txout.